airbyte_cdk/sources/declarative/parsers/model_to_component_factory.py

```diff
@@ -3826,16 +3780,60 @@ def create_substream_partition_router(
     def _create_message_repository_substream_wrapper(
         self, model: ParentStreamConfigModel, config: Config, **kwargs: Any
     ) -> Any:
+        # getting the parent state
+        child_state = self._connector_state_manager.get_stream_state(
+            kwargs["stream_name"], None
+        )  # FIXME adding `stream_name` as a parameter means it will be a breaking change. I assume this is mostly called internally so I don't think we need to bother that much about this but still raising the flag
+        if model.incremental_dependency and child_state:
+            parent_stream_name = model.stream.name or ""
+            parent_state = ConcurrentPerPartitionCursor.get_parent_state(
+                child_state, parent_stream_name
+            )
+
+            if not parent_state:
+                # there are two migration cases: state value from child stream or from global state
+                parent_state = ConcurrentPerPartitionCursor.get_global_state(
+                    child_state, parent_stream_name
+                )
+
+                if not parent_state and not isinstance(parent_state, dict):
+                    cursor_field = InterpolatedString.create(
+                        model.stream.incremental_sync.cursor_field,
+                        parameters=model.stream.incremental_sync.parameters or {},
+                    ).eval(config)
+                    cursor_values = child_state.values()
+                    if cursor_values:
+                        parent_state = AirbyteStateMessage(
+                            type=AirbyteStateType.STREAM,
+                            stream=AirbyteStreamState(
+                                stream_descriptor=StreamDescriptor(
+                                    name=parent_stream_name, namespace=None
+                                ),
+                                stream_state=AirbyteStateBlob(
+                                    {cursor_field: list(cursor_values)[0]}
+                                ),
+                            ),
+                        )
+            connector_state_manager = ConnectorStateManager([parent_state] if parent_state else [])
```

Comment:
worth a note covering why we define a new connector state manager. I don't think I fully understand why? is it so we don't overwrite state when running the parent stream sync?